test/prototype_transforms_kernel_infos.py

```diff
@@ -17,11 +20,24 @@
 __all__ = ["KernelInfo", "KERNEL_INFOS"]
 
 
+TestID = Tuple[Optional[str], str]
+
+
 @dataclasses.dataclass
-class Skip:
-    test_name: str
-    reason: str
-    condition: Callable[[ArgsKwargs, str], bool] = lambda args_kwargs, device: True
+class TestMark:
+    test_id: TestID
+    mark: MarkDecorator
+    condition: Callable[[ArgsKwargs], bool] = lambda args_kwargs: True
+
+
+def mark_framework_limitation(test_id, reason):
+    # The purpose of this function is to have a single entry point for skip marks that are only there, because the test
+    # framework cannot handle the kernel in general or a specific parameter combination.
+    # As development progresses, we can change the `mark.skip` to `mark.xfail` from time to time to see if the skip is
+    # still justified.
+    # We don't want to use `mark.xfail` all the time, because that actually runs the test until an error happens. Thus,
+    # we are wasting CI resources for no reason for most of the time
+    return TestMark(test_id, pytest.mark.skip(reason=reason))
```

Comment:
By changing this to 

```suggestion
    return TestMark(test_id, pytest.mark.xfail(reason=reason))
```
 
temporarily, I actually found a unnecessary skip although we barely have any at the moment. See below for details.